CommonClient.py

```diff
@@ -70,59 +70,123 @@ def _cmd_disconnect(self) -> bool:
+            item_name = self.ctx.item_names[item.item]
+            if not filter_text or filter_text.lower() in item_name.lower():
+                count += 1
+                self.output(f"{item_name} from {self.ctx.player_names[item.player]}")
+        if count == 0:
+            self.output("No received items found")
+        else:
+            self.output(f"Found {count} {'matching ' if filter_text else ''}received items")
         return True
 
-    def _cmd_missing(self, filter_text = "") -> bool:
-        """List all missing location checks, from your local game state.
+    @mark_raw
+    def _cmd_missing(self, filter_text: str = "") -> bool:
+        """List all missing location checks from your local game state.
         Can be given text, which will be used as filter."""
         if not self.ctx.game:
             self.output("No game set, cannot determine missing checks.")
             return False
+        if filter_text:
+            self.output(f"Missing checks matching '{filter_text}':")
+        else:
+            self.output("Missing checks:")
         count = 0
-        checked_count = 0
+        total_count = 0
         for location, location_id in AutoWorldRegister.world_types[self.ctx.game].location_name_to_id.items():
-            if filter_text and filter_text not in location:
+            if filter_text and filter_text.lower() not in location.lower():
                 continue
             if location_id < 0:
                 continue
             if location_id not in self.ctx.locations_checked:
                 if location_id in self.ctx.missing_locations:
                     self.output('Missing: ' + location)
                     count += 1
+                    total_count += 1
+                elif location_id in self.ctx.checked_locations:
+                    total_count += 1
+
+        if count:
+            self.output(f"Found {count}/{total_count} {'matching ' if filter_text else ''}missing checks")
+        else:
+            self.output("No missing checks found")
+        return True
+
+    @mark_raw
+    def _cmd_checked(self, filter_text: str = "") -> bool:
+        """List all checked location checks from your local game state.
+        Can be given text, which will be used as filter."""
+        if not self.ctx.game:
+            self.output("No game set, cannot determine checked checks.")
+            return False
+        if filter_text:
+            self.output(f"Checked locations matching '{filter_text}':")
+        else:
+            self.output("Checked locations:")
+        self.output("")
+        count = 0
+        total_count = 0
+        for location, location_id in AutoWorldRegister.world_types[self.ctx.game].location_name_to_id.items():
+            if filter_text and filter_text.lower() not in location.lower():
+                continue
+            if location_id < 0:
+                continue
+            if location_id not in self.ctx.locations_checked:
```

Comment:
Does this contradict what the documentation says?
It looks like it only displays a location if it's NOT checked according to the local state (and IS checked according to the server state - which was a useful part of the `missing` command before).

Reply:
🤷‍♀️ I copied what `missing` was doing before, which was showing both `Missing:` and `Checked:` items fully, and it works in my tests. Not sure WHY that's the code for that, but, it worked properly for me - as shown in the screenshots provided.

Reply:
You tested with the generic TextClient and Stardew Valley, right?

TextClient isn't the client that controls the information flow between the game and the server, so it doesn't have any "local state" of the game.
That's why this particular `if` condition is always true in your tests - because `self.ctx.locations_checked` isn't used and is always empty.
For any game that uses `self.ctx.locations_checked`, this won't work and is broken.

The `missing` command as it was before you changed it was useful for games with a local state to check the difference between the local state and the server state.
For example:
If I pick up some locations, and then restart the client and the game (or load an earlier save state before I picked up the locations) I could see which locations the server never saw me pick up, and also see which locations the server did see me pick up, but I haven't picked them up again since restarting the game.
Another example:
It's also useful for when another player uses `/collect`. That makes it so the locations are picked up in the server state, but if I want to still pick them up locally, I can see what I haven't picked up from my local state.

Reply:
Could you get what you're looking for from the server commands `!missing` and `!checked`, instead of the client commands `/missing` and `/checked`?

Reply:
> Could you get what you're looking for from the server commands `!missing` and `!checked`, instead of the client commands `/missing` and `/checked`?

Quite possibly. Don't particularly have the energy at the moment to dive into doing more shit with this lol...